### PR TITLE
[Cherry-pick-2.0.4][Improvement-8192][dolphinschedule_sql] Index query performance improvement

### PR DESCRIPTION
**Pull request by ouyangyewei429**
fix: #8192 

Associate with
- PR: #8193

### CODE REVIEW

**Comment by ouyangyewei429** [comment]
@JinyLeeChina @lenboo improvement feature has cherry-pick to 2.0.4-prepare. pls review, tks

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.3_schema/mysql/dolphinscheduler_ddl.sql [hunk 1]
```diff
@@ -15,7 +15,9 @@
  * limitations under the License.
 */
 
+ALTER TABLE `t_ds_task_instance` ADD INDEX `idx_code_version` (`task_code`, `task_definition_version`) USING BTREE;
```
Comment (brave-lee) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.3_schema/mysql/dolphinscheduler_ddl.sql]: Please add this to 2.0.4 schema folder

**Review by lenboo — APPROVED** [review]
+1

**Review by brave-lee — APPROVED** [review]
+1

**Pull request merged**